src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+
+  /**
+   * @brief Get the integer identifier for the first Tribol registered mesh
+   *
+   * @return integer 
+   */
+  integer GetMesh1ID() const { return mesh_id_1_; }
+
+  /**
+   * @brief Get the integer identifier for the second Tribol registered mesh
+   *
+   * @return integer 
+   */
+  integer GetMesh2ID() const { return mesh_id_2_; }
+
+  /**
+   * @brief Get the number of elements on the first mesh
+   * 
+   * @return integer 
+   */
+  integer GetMesh1NE() const
+  { 
+    return GetUpdateData().conn_1_.size() / num_verts_per_elem_;
+  }
+
+  /**
+   * @brief Get the number of elements on the second mesh
+   * 
+   * @return integer 
+   */
+  integer GetMesh2NE() const
+  {
+    return GetUpdateData().conn_2_.size() / num_verts_per_elem_;
+  }
+
+  /**
+   * @brief Get the total number of vertices in both meshes
+   * 
+   * @return integer 
+   */
+  integer GetNV() const { return GetUpdateData().redecomp_.GetNV(); }
+
+  /**
+   * @brief Get the connectivity for the first mesh
+   * 
+   * @return const IndexType* 
+   */
+  const IndexType* GetMesh1Conn() const
+  {
+    return GetUpdateData().conn_1_.data();
+  }
+
+  /**
+   * @brief Get the connectivity for the second mesh
+   * 
+   * @return const IndexType* 
+   */
+  const IndexType* GetMesh2Conn() const
+  {
+    return GetUpdateData().conn_2_.data();
+  }
+
+  /**
+   * @brief Get the element type for the meshes
+   * 
+   * @return integer 
+   */
+  integer GetElemType() const { return elem_type_; }
+
+  /**
+   * @brief Get pointers to component arrays of the coordinates on the redecomp mesh
+   * 
+   * @return std::vector<const real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<const real*> GetRedecompCoordsPtrs() const 
+  { 
+    return coords_.GetRedecompFieldPtrs();
+  }
+
+  /**
+   * @brief Get pointers to component arrays of the nodal response on the redecomp mesh
+   * 
+   * @return std::vector<real*> of length 3
+   *
+   * @note The third entry is nullptr in two dimensions
+   */
+  std::vector<real*> GetRedecompResponsePtrs()
+  {
+    return ParentField::GetRedecompFieldPtrs(redecomp_response_);
+  }
+
+  /**
+   * @brief Get the nodal response grid function on the redecomp mesh
+   * 
+   * @return const mfem::GridFunction& 
+   */
+  const mfem::GridFunction& GetRedecompResponse() const
+  {
+    return redecomp_response_;
+  }
+
+  /**
+   * @brief Get the nodal response grid function on the parent mesh
+   * 
+   * @return mfem::ParGridFunction 
+   */
+  mfem::ParGridFunction GetParentResponse() const
+  {
+    return GetParentTransfer().RedecompToParent(redecomp_response_);
+  }
+
+  /**
+   * @brief Get the parent to redecomp transfer object
+   * 
+   * @return const ParentRedecompTransfer& 
+   */
+  const ParentRedecompTransfer& GetParentTransfer() const
+  { 
+    return GetUpdateData().vector_xfer_;
+  }
+
+  /**
+   * @brief Add/replace the parent velocity grid function
```

Comment:
Set/replace?

Reply:
I think Add is clearer.  `MfemMeshData` can exist without a velocity field, so if a user wants to include one or change the included one, this is the method to call.